python/src/filesystem.rs

```diff
@@ -242,6 +254,35 @@ impl DeltaFileSystemHandler {
             .map_err(PyDeltaTableError::from_object_store)?;
         Ok(file)
     }
+
+    pub fn __getstate__(&self) -> PyResult<Vec<u8>> {
+        // Used in pickle/pickling
+        let s = serde_json::to_string(&self.config).unwrap();
+        Ok(s.as_bytes().to_vec())
+    }
+
+    pub fn __setstate__(&mut self, state: Vec<u8>) -> PyResult<()> {
+        let config: FsConfig = serde_json::from_slice(&state)
+            .map_err(|err| PyDeltaTableError::from_raw(err.into()))?;
+
+        let storage = DeltaTableBuilder::from_uri(config.root_url.clone())
+            .with_storage_options(config.options.clone())
+            .build_storage()
+            .map_err(PyDeltaTableError::from_raw)?;
+
+        self.inner = storage;
+        self.rt = Arc::new(rt()?);
+        self.config = config;
+
+        Ok(())
+    }
+
+    pub fn __getnewargs__(&self) -> PyResult<(String, Option<HashMap<String, String>>)> {
```

Comment:
I wonder if just this method is sufficient. If we define this, do we still need `__setstate__` and `__getstate__`?

Reply:
Initially added this to fix a bug, but you are correct, this si all we need. Removing the rest makes the implementation for pickling quite minimal 😆.